source/use-cases/sensitive-data-encryption.txt

```diff
@@ -124,7 +124,153 @@ In this step, we generate a local master key to expedite setup of our developmen
+            fileBytes = fis.readAllBytes();
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+
+          final byte[] localMasterKey = Arrays.copyOf(fileBytes, 96);
+
+
+Next, specify the KMS provider settings. Since we are using a local master
+key, the provider name is ``local``:
+
+.. tabs::
+
+  tabs:
+
+    - id: java-kms-settings-local
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          String kmsProvider = "local";
+
+          Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>() {{
+             put(kmsProvider, new HashMap<String, Object>() {{
+                 put("key", localMasterKey);
+             }});
+          }};
+
+
+Create the client with the MongoDB connection string and key vault namespace
+(database and collection name):
+
+.. tabs::
+
+  tabs:
+
+    - id: java-keyvault-settings-local
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          String connectionString = "mongodb://localhost:27017";
+          String keyVaultNamespace = "encryption.__keyVault";
+
+          ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder()
+                  .keyVaultMongoClientSettings(MongoClientSettings.builder()
+                          .applyConnectionString(new ConnectionString(connectionString))
+                          .build())
+                  .keyVaultNamespace(keyVaultNamespace)
+                  .kmsProviders(kmsProviders)
+                  .build();
+
+          ClientEncryption clientEncryption = ClientEncryptions.create(clientEncryptionSettings);
+
+
+.. note::
+
+  Ensure that the client has `ReadWrite
+  <https://docs.mongodb.com/manual/reference/built-in-roles/#readWrite>`_
+  permissions to the specified key vault
+  namespace.
+
+Use the client to generate a data encryption key and convert the returned key
+id to base64 for usage in the following "`C. Define a JSON Schema`_" section.
+
+.. tabs::
+
+  tabs:
+
+    - id: java-kms-settings-local
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          BsonBinary dataKeyId = clientEncryption.createDataKey("local", new DataKeyOptions());
+          final String base64DataKeyId = Base64.getEncoder().encodeToString(dataKeyId.getData());
+
+
+Verify that the key was created in your MongoDB replica set in the specified
+key vault namespace. The document should resemble the following:
+
+.. code-block:: javascript
+
+  {
+    "_id" : UUID("..."),
+    "keyMaterial" : BinData(0,"..."),
+    "creationDate" : ISODate("..."),
+    "updateDate" : ISODate("..."),
+    "status" : 0,
+    "masterKey" : {
+      "provider" : "local"
+    }
+  }
+
+`Click here <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-csfle-data-encryption-key/DataEncryptionKeyGenerator.java>`_
+for the source code used to generate the data encryption key in this section.
```

Comment:
nit: [Don’t provide links from ambiguous phrases such as Click here or More information](https://docs.mongodb.com/meta/style-guide/style/links-and-cross-references/)

Reply:
Thanks for pointing that out, will fix.